src/bin/main.rs

```diff
@@ -99,8 +99,11 @@ fn make_opts() -> Options {
     opts.optflag(
         "",
         "check",
-        "Run in 'check' mode. Exits with 0 if input is formatted correctly. Exits \
-         with 1 and prints a diff if formatting is required.",
+        "Run in 'check' mode. Exits with 0 if input is formatted correctly or reading \
+         from stdin. Exits with 1 and prints a diff if formatting is required. When   \
+         reading a diff lines that start with '-' are poorly formatted, lines that    \
+         start with '+' are after formatting, and all other lines are properly        \
+         formatted.",
```

Comment:
I think we may just want to overhaul this altogether to make things more clear, and perhaps differentiate the output vs. exit code. Obviously not this literally, but what about a pivot to something like:

```
"Run in 'check' mode. The exact output and behavior codes vary depending on whether you are running rustfmt against stdin or files.

For file input, rustfmt will exit with 0 if the input is formatted correctly. If formatting changes are required then rustfmt will exit with a 1 and print a diff.

For stdin input, .....

The diff output, lines with +, -, etc.
```

Reply:
and actually thinking about this more, in light of how wordy this could get, it might be worth trying to follow the same model we do with this (a short help text, and then an extended, more detailed version), e.g. `--help=file-lines`, but with `--help=check`

Reply:
I think it's best to keep it brief unless the user specifically asks for it, so the `--help=check` sounds good to me. I will try to keep it brief and then add a "For more info, try `--help=check`".